### PR TITLE
4.0 - Update changes.md file for the new API

### PR DESCRIPTION
**Pull request by CarlosRS9**
Hello team,

This PR is related to [#2413](https://github.com/wazuh/wazuh-documentation/issues/2413) and updates changes.md files to include every major change made to the endpoints in the new API 4.0.

Regards.

### CODE REVIEW

File: api/api/spec/changes.md [hunk 1]
```diff
@@ -1,383 +1,528 @@
...
+    "affected_items": [],
+    "total_affected_items": 0,
+    "total_failed_items": 0,
+    "failed_items": [],
+    "message": ""
+  }
+}
+```
+* In responses, `msg` key is now moved to new `message` key
+* In responses, `filter` key is now moved to new `filters` key
```
Comment (davidjiglesias) [inline api/api/spec/changes.md:21]: Remove this line, I haven`t found any response that uses this key anymore.

File: api/api/spec/changes.md [hunk 2]
```diff
@@ -1,383 +1,528 @@
...
-* Endpoint deleted, use `GET /groups?list_groups=group_id`
-* To get all agents in a group use `GET /groups/{group_id}/agents`
+### DELETE  /agents/{agent_id}
+* Endpoint removed. Use `DELETE /agents?list_agents=agent_id` instead
 
-### GET /agents/groups/{group_id}/configuration
-* Endpoint renamed to `GET /groups/{group_id}/configuration`
-* In response, `filter` key is now moved to new `filters` key
+### DELETE  /agents/{agent_id}/group
+* Added **list_groups** parameter in query to specify an array of group's ID to remove from the agent.
```
Comment (davidjiglesias) [inline api/api/spec/changes.md:46]:
```suggestion
* Added **list_groups** parameter in query to specify an array of group IDs to remove from the agent.
```

File: api/api/spec/changes.md [hunk 3]
```diff
@@ -1,383 +1,528 @@
...
-* Verb changed to POST
+### DELETE  /agents/groups/{group_id}
+* Endpoint removed. Use `DELETE /groups?list_groups=group_id` instead.
 
-### DELETE /agents/:agent_id
-* Endpoint deleted. Use `DELETE /agents?list_agents=agent_id&older_than=0s`
-* Be aware that `DELETE /agents` older_than filter is not defaulted to '0s'
+### GET     /agents
+* Return information about all available agents or a list of them.
+* Added parameter **list_agents** in query used to specify a list of agents IDs (separated by comma) from which agents get the information.
```
Comment (davidjiglesias) [inline api/api/spec/changes.md:64]:
```suggestion
* Added parameter **list_agents** in query used to specify a list of agent IDs (separated by comma) from which agents get the information.
```

File: api/api/spec/changes.md [hunk 4]
```diff
@@ -1,383 +1,528 @@
...
+* Added parameter **registerIP** in query used to filter by the IP used when registering the agent.
+* With this new endpoint, you won't get a 400 response in agent name cannot be found,
```
Comment (davidjiglesias) [inline api/api/spec/changes.md:66]:
```suggestion
* With this new endpoint, you won't get a 400 response if agent name cannot be found,
```

File: api/api/spec/changes.md (hunk 1 above)
Comment (CarlosRS9) [reply api/api/spec/changes.md]: Done

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** aeed3cca6de5f783667b77abff9d8f5d5e238653
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1877/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1877/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1877/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B1877.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1877/Analysis_dynamic_valgrind-v4.0.0-B1877.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** aeed3cca6de5f783667b77abff9d8f5d5e238653
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1878/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1878/ossec.log)
**Coverage:** 

```json
Lines executed:26.49% of 2997

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1878/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B1878.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1878/Analysis_dynamic_valgrind-v4.0.0-B1878.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** aeed3cca6de5f783667b77abff9d8f5d5e238653
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1879/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1879/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1879/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B1879.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1879/Analysis_dynamic_valgrind-v4.0.0-B1879.log)

**Pull request merged**